### ISSUE TITLE
Networking Issue 

### ISSUE THREAD

**Issue by computershateme** [canonical/multipass#3508]
**Describe the bug**
Graphical Interface partially connected to wired ethernet

**To Reproduce**
1. Install KDE desktop 
2. Install NetworkManager
3. Login to Graphical Interface using either X2Go or RDP
4. Open terminal and type multipass networks and see list of available networks
5. Hover over the "Networks" icon on the bottom right of the screen and see "Disconnected"
6. Right click on "Networks" icon and select "Configure Network Connections"
7. Click + sign ("Add new connection") and select "Wired Ethernet"
8. Click dropdown on "Restrict to device" and select a network connection, click "Random" on "Cloned MAC address" line
9. Click "IPv4" tab and put check in box "IPv4 is required for this connection
10. Click "IPv6" tab and put check in box "IPv6 is required for this connection" 
11. Click "Save"
12. Enter sudo systemctl restart NetworkManager && sudo systemctl restart dbus.service && sudo systemctl restart polkit.service"
13. Again click on "Networks" icon and see "No available connections"

**Expected behavior**
- My network listed with the connect option

**Additional info**
- _OS_: Mac Sonoma 14.4.1 (23E224)
- _multipass version_: multipass 1.13.1, multipassd 1.13.1
- _multipass get local.driver_: lxd

**Additional context**
- I am able to access the internet through my web browsers and through terminal, but certain features like "Updates" in the "Software" app say:  "No connection Go online to check for updates"
- mpbr0 is listed when I run the "multipass networks" command but doesn't appear to be active

**Comment by georgeliao** [comment]
Hi @computershateme 
Thanks for reporting. Before we can dig deeper, I think we need more clarification of some information. Your OS is Mac Sonoma and the local driver is lxd, however Multipass does not support lxd on that platform as far as I know. 

Is there something we are missing?

**Comment by computershateme** [comment]
Apologies (I should have double checked which terminal window I was typing the command into), here is the correct information:
- _OS_: Mac Sonoma 14.4.1 (23E224)
- _multipass version_: multipass 1.13.1, multipassd 1.13.1
- _multipass get local.driver_: qemu

**Comment by georgeliao** [comment]
@computershateme 
About the first three steps, are you using a local machine to access the macOS machine?

**Comment by computershateme** [comment]
Hi there. The Mac is my local machine and I am remoting into a plasma desktop I created using multipass. I am able to remote into the desktop using RDP and X2Go. 

Once I am in I can use my web  browsers and Synaptic Package Manager with no problems. When I try to use the "Update" features in the Snap-Store and Software apps it says that there is no network connection. When I open the Network app there are no network connections. When I create one, it doesn't stick.

When I run multipass exec <instance> -- bash -c 'ip -c -br 1'on my Mac, it tells me that the multipass bridge mpbr0 is down and says in the last column <NO-CARRIER, BROADCAST, MULTICAST, UP>.

**Comment by computershateme** [comment]
There's definitely a reason I picked "computers hate me" as my username. 

I am running a Mac mini M1. In Parallels I have Ubuntu, Windows 11, and Arch Linux VM's installed. I also have VMWare Fusion installed and have a Windows 11 machine. In UTM I have Windows 10 ARM, Windows 11, and Manjaro Linux VM's. I have VirtualBox but none of those VM's work. Finally, I have Multipass installed.

The mpbr0 might be from my installing multipass inside one of my instances (I figured that it would be fun to make instances inside instances). 

I'll try removing the multipass from inside the instance and seeing if the mpbr0 disappears...I'll let you know.

**Comment by computershateme** [comment]
Yup, mpbr0 is gone

**Comment by computershateme** [comment]
Ok, so here's where I am now. Just from deleting multipass  that I installed in my instance, the networks are showing up and are showing as connected, but are saying the connection speed is 0 Bit/s.
![Image 4-28-24 at 2 10 AM](https://github.com/canonical/multipass/assets/85035572/682ec21c-750b-4590-9012-18f44784863e)

**Comment by townsend2010** [comment]
Hi @computershateme!

You say they are showing up, so are they working?  The lack of connection speed may just be an artifact of it connecting through a bridged interface.  I will say that it sounds like you are doing things that we have not done :sweat_smile:

**Comment by computershateme** [comment]
Haha I'm all about being different. I feel like we're getting closer because now the networks are showing up in the network area. I'm still able to get to the internet via the web browsers, but the problem remains with the programs that integrate in with the desktop as they are still not connecting with the network.To draw a parallel, I feel like the main water pipe is open but, for some reason, none of the sinks on the second floor have running water.Sent from my iPhoneOn Apr 29, 2024, at 4:02 PM, Chris Townsend ***@***.***> wrote:﻿
Hi @computershateme!
You say they are showing up, so are they working?  The lack of connection speed may just be an artifact of it connecting through a bridged interface.  I will say that it sounds like you are doing things that we have not done 😅

**Comment by townsend2010** [comment]
Hi @computershateme,

Unfortunately, I'm not really sure what else we can do here.  We don't prevent users from installing desktop environments (and we really couldn't if we wanted to), but we don't provide any official support for that.  You seem to be in uncharted territory :slightly_smiling_face: 

I'm going to close this, but feel free to reopen or open a new issue if you can determine that something we're doing in Multipass is at fault.  Sorry we couldn't be of more help.

**Comment by computershateme** [comment]
I am wondering if it has something to do with the bridge and the way multipass is treating each session. 

When I multipass shell or ssh into an instance and run loginctl list-sessions, I see a bunch of sessions...even ones that aren't being used any longer. I will then clean that list using loginctl terminate-user.

Once I know that all sessions are closed, I will ssh -X into the user instance and will see a similar problem even with just -X forwarding.

**Comment by townsend2010** [comment]
The sessions you see are due to the tray icon GUI having to poll the daemon to get near real time info about instances which SSH to get this info.  It's not ideal, but it works and the upcoming 1.14 release makes this much, *much* better.

**Comment by townsend2010** [comment]
I will say it seems like networking works except for some apps in the desktop session, but other apps that use the network will work.  I don't know the internals of the desktop environment enough to venture a guess at what it is missing.

**Comment by computershateme** [comment]
Then I'm super excited for 1.14 :-) 

I'll try the Ubuntu board to see if they know how to clear the desktop issue.

Thanks for your help
